lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -170,23 +170,25 @@ def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
         end
       end
 
-      # Update instance list
+      # Update instance list, but only if there are machines specified
       actual_instance_ids = Set.new(actual_elb.instances.map { |i| i.instance_id })
 
-      instances_to_add = machine_specs.select { |s| !actual_instance_ids.include?(s.location['instance_id']) }
-      instance_ids_to_remove = actual_instance_ids - machine_specs.map { |s| s.location['instance_id'] }
-
-      if instances_to_add.size > 0
-        perform_action.call("  add machines #{instances_to_add.map { |s| s.name }.join(', ')}") do
-          instance_ids_to_add = instances_to_add.map { |s| s.location['instance_id'] }
-          Chef::Log.debug("Adding instances #{instance_ids_to_add.join(', ')} to load balancer #{actual_elb.name} in region #{aws_config.region}")
-          actual_elb.instances.add(instance_ids_to_add)
+      if machine_specs
```

Comment:
Do you have the latest chef-provisioning?  I ask because I think I fixed this issue on the caller side: https://github.com/chef/chef-provisioning/blob/master/lib/chef/provider/load_balancer.rb#L28